scripts/irbrc.rb

```diff
@@ -73,3 +38,40 @@
 rescue => exception
   puts "Error : 'load #{ENV["HOME"]}/.irbrc' : #{exception.message}"
 end
+
+# Turn on completion.
+begin
+  require "readline"
+
+  require "irb/completion" rescue nil
+
+  # Turn on history saving.
+  # require "irb/ext/save-history"
+  # IRB.conf[:HISTORY_FILE] = File.join(ENV["HOME"], ".irb-history")
+
+  # Use an alternate way to on history saving until save-history is fixed.
+  #
+  #   bug:   http://redmine.ruby-lang.org/issues/show/1556
+  #   patch: http://pastie.org/513500
+  #
+  # This technique was adopted from /etc/irbrc on OS X.
+  histfile = File.expand_path(".irb-history", ENV["HOME"])
+  histfile = IRB.conf[:HISTORY_FILE] || histfile
```

Comment:
For the record, I'm working on an ubuntu box. I need to set `IRB.conf[:HISTORY_FILE]` and `IRB.conf[:SAVE_HISTORY]` to get irb history to save (meaning all the other readline/irb history code is removed). For me, readline works without `require readline`, as well, which means on my machine, I can boil the readline/irb history code down to 2 lines!

One minor note, though... There was code that eliminated duplicate lines from being saved in the history. The 'default' behavior appears to be that every line gets saved even if it's a duplicate, but I don't think that's a big enough reason to keep all that code.